app/src/main/java/com/example/hackernews/viewmodel/MainViewModel.kt

```diff
@@ -30,13 +34,27 @@ import com.example.hackernews.model.repository.NewsRepository
  * Having said that, ViewModel will be responsible for fetching stories and all the data required by the activity and
  * it will only expose data via observable properties. We'll use live data for for this purpose.
  */
-class MainViewModel(private val newsRepository: NewsRepository) : ViewModel() {
+
+class MainViewModel(
+    private val newsRepository: NewsRepository,
+    private val userRepository: UserRepository
+) : ViewModel() {
     private val _topStories = arrayListOf<News>()
```

Comment:
Have variable for every kind of news, `topStories , latestStories, savedStories` etc and based on user selection update news live data object `news.value = savedStories`